### ISSUE TITLE
A FAQ for common problems

### ISSUE THREAD

**Issue by Tenchi2xh** [o5k/tetrio-issues#241]
(Could be a markdown file in this repo or this issue stickied? Also maybe pinned in the chat and in `#resources`)

Goal: to avoid users encountering common issues from repeating the same things in the chat or issues, and provide them with a quick link to solve their problems

I can see two such issues already:

EDIT: adding more questions gradually, the two initial ones that prompted this were about wallpapers and cache

The TOC will only work as a markdown file and not an issue

# FAQ

- [General](#general)
    - [My wallpaper won't load](#my-wallpaper-wont-load)
    - [My game is behaving weirdly / I can't connect / I keep getting connection errors](#my-game-is-behaving-weirdly--i-cant-connect--i-keep-getting-connection-errors)
- [Gameplay](#gameplay)
    - [How many lines of garbage do my attacks send?](#how-many-lines-of-garbage-do-my-attacks-send)
- [Tetra League](#tetra-league)
    - [I have finished my 10 placement matches, but still have no rank?](#i-have-finished-my-10-placement-matches-but-still-have-no-rank)
    - [How does the game decide which rank I have?](#how-does-the-game-decide-which-rank-i-have)
- [Glossary of abbreviations](#glossary-of-abbreviations)

### My wallpaper won't load

This is probably due to the image host where the wallpaper was uploaded not supporting [CORS Headers](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing), meaning that the host does not explicitly allow web applications to download their content.

Solutions:

- Upload the wallpaper on another image host
- Use a CORS proxy, which will get around the header issue by adding their own. For this, just add `https://cors-anywhere.herokuapp.com/` in front of the URL of the wallpaper (for example, `https://cors-anywhere.herokuapp.com/https://foobar.com/my_blocked_wallpaper.png`

### My game is behaving weirdly / I can't connect / I keep getting connection errors

This is probably due to an old cache. Solutions:

- Do a forced refresh: <kbd>Ctrl</kbd>+<kbd>F5</kbd>, or <kbd>⌘ Cmd</kbd>+<kbd>⇧ Shift</kbd>+<kbd>R</kbd> on Mac
- Clear the cache of your browser and reload the game

### How many lines of garbage do my attacks send?

Here is the current attack table used by the game, showing how many lines of garbage an attack will send, depending on when in a combo the attack was made:

![Attack table](https://cdn.discordapp.com/attachments/673303546564968566/697422669032980490/2020-04-08_14-28-20.png)

For example, doing the combo `Triple` → `Single` → `Single` → `Quad` will send 2 + 0 + 1 + 7 = 10 lines of garbage total.

### I have finished my 10 placement matches, but still have no rank?

You will receive a rank when your RD (the uncertainty score, the part after the ±) is under 100

<img src="https://user-images.githubusercontent.com/4116708/78863504-0c396000-7a3a-11ea-8151-64162de777af.png" width=200>

For that all you need to do is play more games until the system is sure about your actual ranking. The placement matches are only there to give you an initial position

### How does the game decide which rank I have?

Here is the current rank distribution used by the game:

![Ranks cheat sheet](https://cdn.discordapp.com/attachments/696095920248455229/696449895132954644/league-cheatsheet.png)

This means that the ranks are distributed equally across the player base and is not intrinsically tied to the exact number of TR or Glicko points — the ranks are spread throughout the whole player base, meaning that the top 1% will always have the rank X while the bottom 3.5% will always have the rank D.

In other words, if the server ever only contains pro-level players, the bottom 3.5% of those pro-level players will have the rank D. And similarly, if the server only ever contains beginner-level players, the top 1% of those would have the rank X.

Moreover, your rank will only get updated if you play league games, and can only go down if you lose, and only go up if you win.

## Glossary of abbreviations

TODO: Complete

<dl>
    <dt>PPS</dt>
    <dd>Pieces Per Second</dd>
    <dt>APM</dt>
    <dd>Attacks Per Minute</dd>
    <dt>TR</dt>
    <dd>Tetr.io Ranking</dd>
    <dt>PC</dt>
    <dd>Perfect Clear</dd>
    <dt>SD</dt>
    <dd>Soft Drop</dd>
    <dt>...</dt>
    <dd>...</dd>
</dl>

**Comment by Tenchi2xh** [comment]
More questions to add:

- What is finesse?
- How is the VS score calculated? How do I improve my VS score?
- How are TR, glicko and RD calculated?

**Comment by Tenchi2xh** [comment]
I implemented a fan-FAQ on my fansite: https://tetrio.team2xh.net/?t=faq
